[Ashea10] iP

ashea10 opened the pull request:
# ConverSage: Your Smart Task Manager

** ConverSage ** is more than just a mere chatbot—it's your very own personal assistant, designed to keep your life organized and stress-free. Here's why you'll love it:

- **Intuitive and user-friendly** interface for seamless interaction
- **Lightning fast responses** to keep up with your busy schedule
- **Completely free**, with no hidden costs or subscriptions

Get started in seconds:

1. Download ConverSage from <u> [**our website**](https://dictionary.cambridge.org/dictionary/english/hello)
2. Launch the application with a simple double-click.
3. Add your tasks and watch ConverSage work its magic ✨


## Features:

- **Task Management**: Effortlessly create, update, and delete tasks.
- **Deadline Tracking**: Never miss a due date again.
- **Custom Reminders** (coming soon): Set reminders for tasks and appointments.

Java programmers, ConverSage is a fantastic way to hone your skills. Here's a sneak peek of the main method:

``` Java
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

### Your Todo-List
- [X] Download Conversage
- [X] Launch the application
- [ ] Add your tasks
- [ ] Relax and let ConverSage keep you on track!

> **_"The bad news is time flies. The good news is you’re the pilot." – Michael Altshuler_**

### Some Comands

- Use `list` to list down all your tasks
- Use `find` to find a specific task

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

ashea10 committed:
Modify .gitignore file

ashea10 committed:
Implement lvl 0

ashea10 committed:
Implement level-1

ashea10 committed:
Add Task.java

ashea10 committed:
Implement Level 3

ashea10 committed:
Add Task subclasses

ashea10 committed:
Implement level-4

ashea10 committed:
Implement automated testing

ashea10 committed:
Add exception classes

ashea10 committed:
Implement error-handling

ashea10 committed:
Task class: Fix bug in markAsUndone method

ashea10 committed:
Implement task delete

ashea10 committed:
Update Task class and subclasses to handle saving

ashea10 committed:
Add data folder to .gitignore

ashea10 committed:
Implement level-7 features

ashea10 committed:
Merge branch 'branch-Level-7'

ashea10 committed:
Add command to list tasks on specific date

ashea10 committed:
Implement level 8

ashea10 committed:
Merge branch 'branch-Level-8'

ashea10 committed:
Implement Storage class to manage loading and saving tasks

ashea10 committed:
Implement Ui class to handle user interactions

ashea10 committed:
Implement Parser class to handle user command

ashea10 committed:
Implement TaskList class to handle task operations

ashea10 committed:
Implement Command class and subclasses

ashea10 committed:
Implement A-MoreOOP

ashea10 committed:
Merge branch 'branch-A-MoreOOP'

ashea10 committed:
Refactor code to use packages

ashea10 committed:
Merge branch 'branch-A-Packages'

ashea10 committed:
Merge branch 'add-gradle-support'

ashea10 committed:
Add JUnit tests

ashea10 committed:
Merge branch 'branch-A-JUnit'

davidgohzk commented on src/main/java/conversage/ConverSage.java:
```diff
@@ -0,0 +1,48 @@
+import conversage.task.TaskList;
+import conversage.ui.Ui;
+
+public class ConverSage {
```
I noticed that there are a lot of places where there is a lack of javadocs...

davidgohzk commented on src/main/java/conversage/storage/Storage.java:
```diff
@@ -0,0 +1,65 @@
+import conversage.exception.ConverSageException;
+import conversage.task.Task;
+
+import java.io.BufferedReader;
```
please check the correct import order for CS2103. You can find the configuration here: https://se-education.org/guides/tutorials/intellijCodeStyle.html

davidgohzk commented on src/main/java/conversage/task/Task.java:
```diff
@@ -0,0 +1,75 @@
+        boolean isDone = parts[1].equals("Done");
+        String taskDesc = parts[2];
+
+        switch (taskType) {
```
Review the coding style for switch case: https://se-education.org/guides/tutorials/intellijCodeStyle.html

davidgohzk commented on src/main/java/conversage/ui/Ui.java:
```diff
@@ -0,0 +1,47 @@
+    }
+
+    public void showWelcome() {
+        String logo = "  ____                          ____                   \n" +
```
`+` should be on a new line

davidgohzk reviewed (commented):
I like that you make good use of custom exceptions and I can see it through the project, so good job on that!

Do review the coding style for this module also!

ashea10 committed:
Fix directory creation bug in Storage.java

ashea10 committed:
Update build.gradle

ashea10 committed:
Implement FindCommand class

ashea10 committed:
Add JavaDoc comments for all methods/classes

ashea10 committed:
Update code to comply with coding standard

ashea10 committed:
Merge branch 'branch-Level-9'

ashea10 committed:
Merge branch 'branch-A-JavaDoc'

ashea10 committed:
Resolve merge conflict

jolwnn commented on src/main/java/conversage/Parser.java:
```diff
@@ -0,0 +1,64 @@
+        String[] commandParts = fullCommand.split(" ", 2);
+        String command = commandParts[0];
+
+        switch (command.toLowerCase()) {
```
I like that you used the correct indentation for switch here!

jolwnn commented on src/main/java/conversage/ConverSage.java:
```diff
@@ -0,0 +1,66 @@
+     */
+    public void run() {
+        ui.showWelcome();
+        boolean isExit = false;
```
I like that you followed the Java Coding Standards naming for booleans here!

jolwnn commented on src/main/java/conversage/command/ListCommand.java:
```diff
@@ -0,0 +1,25 @@
+ */
+public class ListCommand extends Command {
+    public ListCommand() {
+    }
```
I'm not sure why is there an empty method?

jolwnn commented on src/main/java/conversage/task/Deadline.java:
```diff
@@ -0,0 +1,44 @@
+public class Deadline extends Task{
+    protected LocalDateTime deadline;
+    protected static final DateTimeFormatter INPUT_FORMAT = DateTimeFormatter.ofPattern("yyyy-MM-dd HHmm");
+    protected static final DateTimeFormatter OUTPUT_FORMAT = DateTimeFormatter.ofPattern("MMM d yyyy, h:mma");
```
I like that the order of the method modifiers follows the coding standards, as per https://se-education.org/guides/conventions/java/index.html#methods

jolwnn commented on src/main/java/conversage/task/Deadline.java:
I like that the order of the declarations of variables and methods in these classes follow the java coding standards, as per https://se-education.org/guides/conventions/java/index.html#classes-and-interfaces.

jolwnn reviewed (commented):
LGTM, the names of the files are packages and classes are ok

ashea10 committed:
Add config files for checkstyle

ashea10 committed:
Update build.gradle for checkstyle

ashea10 committed:
Merge branch 'branch-A-CheckStyle'

ashea10 committed:
Remove unused dependency

ashea10 committed:
Update dependencies for JavaFX in gradle

ashea10 committed:
Modify Command class to return string for GUI

ashea10 committed:
Add getResponse method in ConverSage class

ashea10 committed:
Update build.gradle for GUI

ashea10 committed:
Implement GUI

ashea10 committed:
Merge branch 'branch-Level-10'

ashea10 committed:
Add assert statements to various classes

Added assert statements to ensure assumptions hold true at various
points in the code. This helps in making the code more robust and
easier to debug.

Changes include:
- DeleteCommand: Ensure task index is within valid range
- FindCommand: Ensure task list is not null
- MarkCommand: Ensure task index is within valid range
- Deadline: Ensure deadline is parsed correctly
- TaskList: Ensure task is not null and index is within valid range
- Ui: Ensure message is not null

These assertions help catch potential issues early and improve code
quality.

ashea10 committed:
Remove trailing whitespace and other coding violations

Several files had trailing whitespace and other coding violations that
needed to be addressed to comply with the CS2103T's coding standards.

The following changes were made:
- Removed trailing whitespace from multiple lines
- Ensured proper line breaks for string concatenation
- Fixed import order in some files to comply with the style guide
- Corrected method naming conventions

These changes improve code readability and maintainability, ensuring
that the codebase adheres to the project's coding standards.

ashea10 committed:
Merge pull request #1 from ashea10/branch-A-Assertions

Add assert statements to various classes

ashea10 committed:
Merge branch 'master' into branch-A-CodeQuality

ashea10 committed:
Merge pull request #2 from ashea10/branch-A-CodeQuality

Remove trailing whitespace and other coding violations

ashea10 committed:
Implement help command

ashea10 committed:
Merge branch 'branch-C-Help'

ashea10 committed:
Handle input when command is given without argument

ashea10 committed:
Bye command now exits application

ashea10 committed:
Merge branch 'branch-A-MoreErrorHandling'

ashea10 committed:
Enable automatic resizing of window

ashea10 committed:
Merge branch 'branch-A-BetterGui'

ashea10 committed:
Add Ui.png

ashea10 committed:
Add user guide

ashea10 committed:
Merge branch 'branch-A-UserGuide'

ashea10 committed:
Change user and duke images

ashea10 committed:
Handle empty list message

ashea10 committed:
Personalize help message

ashea10 committed:
Fix minor bug

ashea10 committed:
Add personality

ashea10 committed:
Add favicon and GUI title

ashea10 committed:
Merge branch 'branch-A-Personality'

ashea10 committed:
Update Ui.png

ashea10 committed:
Add missing JavaDoc comments
